### PR TITLE
BUG: Fixed Categorical.Equals with unordered

### LINKED ISSUE #16603: CategoricalIndex.equals incorrectly considers category order when unordered
```python
>>> import pandas as pd
>>> pd.CategoricalIndex(['a', 'b'], categories=['a', 'b']).equals(
...     pd.CategoricalIndex(['a', 'b'], categories=['b', 'a']))
False
```

I fixed this for regular Categoricals in https://github.com/pandas-dev/pandas/pull/16339, but that didn't affect CategoricalIndex.equals. Do we want `.equals` to ignore order when `ordered=False` on both (I think we do)?

Discovered during my CategoricalDtype refactor, which does fix it so that order is ignored when `order=False`.

### PR DESCRIPTION
**Pull request by TomAugspurger**
The original issue was already fixed. I added tests to verify (but no whatsnew entry).

This addes tests and a fix for https://github.com/pandas-dev/pandas/issues/16603#issuecomment-349290078 about `Categorical.equals`

Closes https://github.com/pandas-dev/pandas/issues/16603

### CODE REVIEW

File: pandas/core/categorical.py [hunk 1]
```diff
@@ -2092,8 +2092,16 @@ def equals(self, other):
         -------
         are_equal : boolean
         """
-        return (self.is_dtype_equal(other) and
-                np.array_equal(self._codes, other._codes))
+        if self.is_dtype_equal(other):
+            if self.categories.equals(other.categories):
+                # fastpath to avoid re-coding
+                return np.array_equal(self._codes, other._codes)
```
Comment (jreback) [inline pandas/core/categorical.py]: could make this slightly simper

```
if self.categories.equals(....):
      other_codes = other._codes
else:
     other_codes = _recode......
return np.array_equal(self._codes, other_codes)
```

**Review by jreback — CHANGES_REQUESTED** [review]
small code comment.

File: doc/source/whatsnew/v0.22.0.txt [hunk 2]
```diff
@@ -1,14 +0,0 @@
-.. _whatsnew_0220:
-
-v0.22.0
--------
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: revert this

File: doc/source/whatsnew/v0.23.0.txt [hunk 3]
```diff
@@ -365,7 +365,7 @@ Numeric
 Categorical
 ^^^^^^^^^^^
 
--
+- Bug in ``Categorical.equals`` between two unordered categories with the same categories, but in a different order (:issue:`16603`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: use ``:func:`` here

**Review by jreback — CHANGES_REQUESTED** [review]
just a doc change, otherwise lgtm.

File: doc/source/whatsnew/v0.23.0.txt (hunk 3 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.23.0.txt]: We don't list all the `Categorical` methods in the API docs, and `.equals` is one of the excluded ones.

**Comment by jreback** [comment]
rebase and push again, fixed some hanging by Travis CI

**Pull request merged**